### ISSUE TITLE
XML Strategy returns double tags

### ISSUE THREAD

**Issue by pietercolpaert** [iRail/The-DataTank#134]
When I try to use the generic XML resource, the xml formatters returns everything correctly. The .about and the .json are very strange (double tag names)

**Comment by coreation** [comment]
Hej pieter!

I understand the problem, and I'm trying to find a solution but let me explain first why this happens.

Let's say we have an xml file books

[ books ]
  [ book id=101 ]
      [ author ] John Doe [/author]
  [/book]
[/books]

What this piece of xml tells us is that book is an object with a datamember id and a child object author. Thus in order to not loose information about this structure we process this to the php object model as follows:

stdClass (being book) has a public datamember "id" and a childobject called author. 

This author object has a value, namely John Doe. Since we don't know how to name this child, as it's only a text field, we use the same name as the object namely author. This causes the double formatting in json for example.

I'm going to try to find out what I can do to make the model compatible for both an xml representation ( who direly needs the way we read XML files now, so now information lost ) and another representation.

If you look at it though, say a json representation, it shows that the child objects of a certain object, are indeed objects, and not values, so in a way it shows it correctly but ofcourse it's not the ideal way :).

Hope this clears some fog on how/why this "error" occurs.

**Comment by lievenjanssen** [comment]
why do you need to create a child element for the "John Doe" string literal?

btw the xml example is not displayed as it should (maybe you could use square brackets [] instead of smaller/greater then <>)

**Comment by coreation** [comment]
Fixed, so author is a subobject of book, not an attribute. Translation to the php object model ( Jeroen agrees with this as well afaik ) is that an attribute is datamember, and author is another object. Otherwise you don't make a difference between an attribute and a subobject, which will lead to a non-equal xml representation of the input xml.

**Comment by coreation** [comment]
TL;DR : XML output will now be the same as the input, besides that attributes will now contain an underscore as first character. Non-XML output will contain the sub-objects of xml-objects as literals and not as objects if they contain only a text value.
